Source/DataSources/PointVisualizer.js

```diff
@@ -204,29 +291,77 @@ define([
                     items.set(entity.id, new EntityData(entity));
                 }
             } else {
-                returnPointPrimitive(items.get(entity.id), unusedIndexes);
+                returnPrimitive(items.get(entity.id), unusedPointIndexes, unusedBillboardIndexes);
                 items.remove(entity.id);
             }
         }
 
         for (i = removed.length - 1; i > -1; i--) {
             entity = removed[i];
-            returnPointPrimitive(items.get(entity.id), unusedIndexes);
+            returnPrimitive(items.get(entity.id), unusedPointIndexes, unusedBillboardIndexes);
             items.remove(entity.id);
         }
     };
 
-    function returnPointPrimitive(item, unusedIndexes) {
+    function returnPrimitive(item, unusedPointIndexes, unusedBillboardIndexes) {
         if (defined(item)) {
             var pointPrimitive = item.pointPrimitive;
             if (defined(pointPrimitive)) {
                 item.pointPrimitive = undefined;
                 pointPrimitive.id = undefined;
                 pointPrimitive.show = false;
-                unusedIndexes.push(pointPrimitive._index);
+                unusedPointIndexes.push(pointPrimitive._index);
+                return;
+            }
+            var billboard = item.billboard;
+            if (defined(billboard)) {
+                item.billboard = undefined;
+                billboard.id = undefined;
+                billboard.show = false;
+                unusedBillboardIndexes.push(billboard._index);
             }
         }
     }
 
+    function createCallback(centerAlpha, cssColor, cssOutlineColor, cssOutlineWidth, newPixelSize) {
```

Comment:
No. Does it look very similar to somewhere else? It was removed when we added the point primitive. We had to add it back in for points on terrain since we didn't want to bloat the `PointPrimitiveCollection`.